src/v/cluster/scheduling/topic_memory_per_partition_default.h

```diff
@@ -0,0 +1,32 @@
+/*
```

Comment:
Let us put an INFO-level log line at startup that exposes key info about partition memory calculation, etc? Like you have X memory and this Y p memory setting, and Z shares and W total shares, and parts_per_shard=V so you get this many partitions total.

Reply:
Sure. Did you put this comment into this file intentionally?

Bit confused as you mention "Z shares and W total shares" but the partition side doesn't use any shares so wondering whether you think we should print only the partition relevant info (maybe in the partition allocator?) or also all the other memory group info (including shares for other subsystems).

Reply:
@StephanDollberg - I intentionally put it a (essentially arbitrary) file because comments in the top level PR seem to get lost? They are treated different than review comments, don't have threaded conversations, etc. Maybe I could just have a conventon of using the first line of the first file, dunno.

Reply:
> Bit confused as you mention "Z shares and W total shares" 

Partitions have 10 shares in memory groups that's what I was referring to. So my comment WXYZ thing was really about memory groups: we should print all the "variable info" from memory groups so we know exactly how the memory was split.

We should also have good logging about partition allocations decisions (and failure reasons), but I think we already have that?